### DISCUSSION TITLE
Custom Protocols in open-actions.conf

### DISCUSSION THREAD

**Discussion by mil-ad** [kovidgoyal/kitty#5536] [Q&A] [answered]
Does kitty support handling custom protocols in its mouse scripting settings?

I'm trying to make kitty understand `gs://foo/bar` links so that I can click and open them in a browser. I added a `open-actions.conf` file with the content below in my `~/.config/kitty` but it doesn't do anything.

```
protocol gs
action echo $1 | awk -F '//' '{print "https://console.cloud.google.com/storage/browser/"$2}' | xargs open
```

**Comment by kovidgoyal** [comment, marked as answer]
Yes it does, your action specification is invalid. Read the docs at
https://sw.kovidgoyal.net/kitty/open_actions/

**Reply by mil-ad** [reply]
Happy that it's supported! 🙏🏼 

Could you point me to the bit that's invalid? I thought the action could be wrong but I tried changing it to something from the docs (`action change_font_size current -2`) but the `gs://` links are still not highlighted. I guess `protocol` is not enough then? Do I also need to define `url` or is that optional?

**Reply by kovidgoyal** [reply]
First of all, you need to generate actual terminal hyperlinks not output gs://whatever text. Then if you want to run a command the action is action launch --type=background sh -c "whatever"

**Reply by mil-ad** [reply]
okay that's useful to know. I thought it worked like http/https that are recognized even with a `echo http://whatever`

**Reply by kovidgoyal** [reply]
For that you use the hints kitten or you set url_prefixes in kitty.conf

**Reply by mil-ad** [reply]
amazing!
